Source/WebCore/dom/Document.cpp

```diff
@@ -614,6 +621,21 @@ Document::Document(Frame* frame, const Settings& settings, const URL& url, Docum
 
     InspectorInstrumentation::addEventListenersToNode(*this);
     setStorageBlockingPolicy(m_settings->storageBlockingPolicy());
+        
+#if ENABLE(NOISY_APIS)
+    m_identifierProvider->setShouldUseNoisyApis(page()->settings().noisyAPIsEnabled());
+    // FIXME: Only call for mainframe.
+    if (isHTMLDocument() && !m_url.isEmpty()) {
```

Comment:
If settings().noisyAPIsEnabled() returns false, we don't want to do what is inside this if statement.